android/src/main/kotlin/com/viam/ble/Peripheral.kt

```diff
@@ -185,13 +186,20 @@ class Peripheral(
                 status: Int,
                 newState: Int,
             ) {
-                Log.d(TAG, "onConnectionStateChange $status $newState")
+                Log.d(TAG, "onConnectionStateChange status: $status state: $newState device: ${device.name} ${device.address}")
                 if (newState == BluetoothProfile.STATE_CONNECTED) {
                     requireNotNull(gatt)
                     gatt.discoverServices()
                 } else {
                     disconnected = true
-                    connectedContinuation?.resumeWithException(Exception("failed to connect with GATT status: $status state: $newState"))
+                    val exceptionStr = "failed to connect to device: ${device.name} ${device.address} with GATT status: $status state: $newState"
+                    // 147 corresponds to GATT_CONNECTION_TIMEOUT but that was introduced in API level 35, so doing a straight
```

Comment:
What is `GATT_CONNECTION_TIMEOUT` and API level 35? Just a little confused what this comment means.

Reply:
based on https://developer.android.com/reference/android/bluetooth/BluetoothGatt#GATT_CONNECTION_TIMEOUT, I understood the `Added in API level 35` to mean that the constant was not added until API level 35, so since we're on min API level 29, GATT_CONNECTION_TIMEOUT does not compile. I can see the comment being confusing tho, let me tweak it

Reply:
Gotcha; looks a bit better to me now thank you.